kube/stateful_set_test.go

```diff
@@ -59,43 +51,42 @@ func TestStatefulSetPorts(t *testing.T) {
 	if !assert.NotNil(portDef) {
 		return
 	}
+
 	statefulset, deps, err := NewStatefulSet(role, &ExportSettings{})
 	if !assert.NoError(err) {
 		return
 	}
-	var endpointService, headlessService *apiv1.Service
 
-	if assert.Len(deps.Items, 3, "Should have three services per stateful role") {
-		for _, item := range deps.Items {
-			svc := item.Object.(*apiv1.Service)
-			if svc.Spec.ClusterIP == apiv1.ClusterIPNone {
-				headlessService = svc
+	var endpointService, headlessService helm.Node
+	items := deps.Get("items").Values()
+	if assert.Len(items, 3, "Should have three services per stateful role") {
```

Comment:
Huh, we have 3 items, but only look at two of them?
Also, why are we doing the explicit check here, when we do the full serialized YAML check later anyway?

Reply:
It was like this in the old code. But since it was easy to extend I checked the 3rd item as well in https://github.com/SUSE/fissile/commit/345a015c2d75aec4a02d8409ad458793f308aa83